site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -0,0 +1,223 @@
+                    continue
+        return stream_entries
+
+    def _handle_existing_records(
+        self, entry, existing_rec_id, existing_records_hits, inspire_id
+    ):
+        existing_entries = existing_records_hits[0]["files"]["entries"]
+        new_entries = entry["files"].get("entries", {})
+
+        # Normalize the checksum format in existing for comparison
+        existing_checksums = {
+            key: value["checksum"].replace("md5:", "")
+            for key, value in existing_entries.items()
+        }
+        new_checksums = {key: value["checksum"] for key, value in new_entries.items()}
+
+        # Find key-value pairs in new entry but not in existing
+        only_in_entry = {
+            key: value
+            for key, value in new_entries.items()
+            if new_checksums[key] not in existing_checksums.values()
+        }
+
+        # Find file keys in existing but not in new entry
+        only_in_existing = [
+            key
+            for key, value in existing_entries.items()
+            if existing_checksums[key] not in new_checksums.values()
+        ]
+
+        differences = len(only_in_entry) + len(only_in_existing)
+        if differences:
+            new_version_success = self._create_new_version(
+                only_in_entry, only_in_existing, inspire_id, entry, existing_rec_id
+            )
+            if not new_version_success:
+                return False
+        else:
+            current_rdm_records_service.edit(system_identity, existing_rec_id)
+            current_rdm_records_service.update_draft(
+                system_identity, existing_rec_id, entry
+            )
+            current_rdm_records_service.publish(system_identity, existing_rec_id)
+
+    def _create_new_version(
+        self, only_in_entry, only_in_existing, inspire_id, entry, existing_rec_id
+    ):
+        """For records with updated files coming from INSPIRE, create and publish a new version."""
+        new_version_draft = current_rdm_records_service.new_version(
+            system_identity, existing_rec_id
+        )
+        draft_id = new_version_draft.to_dict()["id"]
+        files_enabled = new_version_draft.to_dict()["files"]["enabled"]
+
+        if files_enabled:
+            current_rdm_records_service.import_files(system_identity, draft_id)
+
+        if only_in_entry:
+            try:
+                if not files_enabled:
+                    draft = new_version_draft._record
+                    draft.files.enabled = True
+                    draft.commit()
+
+                self._create_files(only_in_entry, draft_id, inspire_id)
+            except WriterError:
+                current_rdm_records_service.delete_draft(system_identity, draft_id)
+                return False
+
+        if only_in_existing:
+            for key in only_in_existing:
+                current_rdm_records_service.draft_files.delete_file(
+                    system_identity, draft_id, key
+                )
```

Comment:
Isntead of importing and creating the ones from the entry and deleting the ones we imported and are not there, can we just simply create the ones being harvested? (similar to how is done in _create_new_record)